### PR TITLE
python310Packages.sphinxcontrib-confluencebuilder: init at 1.8.0

### PR DESCRIPTION
**Pull request by graysonhead**
###### Description of changes

This is a sphinxcontrib module for Sphinx that allows it to create/update a document within confluence.

For some reason buildPythonPackage doesn't like the way the unit tests are structured within this project, so I've disabled them. All functionality appears to work fine.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by fufexan — CHANGES_REQUESTED** [review]
Please run `nixpkgs-fmt` over the file. Also, add yourself as a maintainer.

I assume `platforms` is handled by `buildPythonPackage`? If that's not the case, please add that as well.

**Comment by graysonhead** [comment]
@fufexan I added myself as a maintainer and ran nixfmt. I haven't added platforms for the other Python packages I've added so I suppose it handles the platform bit? I'm not exactly sure.

**Comment by winterqt** [comment]
> I assume `platforms` is handled by `buildPythonPackage`? 

Correct.

@graysonhead You don't need to worry about it.

**Comment by winterqt** [comment]
@ofborg build python310Package.sphinxcontrib-confluencebuilder

**Review by winterqt — CHANGES_REQUESTED** [review]
Please fix your commit message to fix the typo in the attribute name -- `python310Package` -> `python310Packages`.

**Comment by graysonhead** [comment]
Whoops, fixed.

File: pkgs/development/python-modules/sphinxcontrib-confluencebuilder/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  };
+
+  propagatedBuildInputs = [
+    docutils
+    sphinx
+    requests
+    jinja2
+  ];
+
+  # Something about the way the unit tests are structured in this project is causing them to fail
```
Comment (winterqt) [inline pkgs/development/python-modules/sphinxcontrib-confluencebuilder/default.nix:26]: I got tests running, but they have a circular dependency on Sphinx, which seems... messy. Per [this package](https://github.com/NixOS/nixpkgs/blob/180147d3dc9df04abc958e15417451096d95a414/pkgs/development/python-modules/sphinxcontrib-serializinghtml/default.nix#L17), we should do the same here.

```suggestion
  # Tests are disabled due to a circular dependency on Sphinx
```

**Review by winterqt — APPROVED** [review]

**Pull request merged**